server/apps/acсount/models.py

```diff
@@ -0,0 +1,200 @@
+
+    def save(self, *args: Any, **kwargs: Any) -> None:
+        """Метод для сохранения изображения."""
+        super().save(*args, **kwargs)
+        if self.avatar:
+            try:
+                image = Image.open(self.avatar.path)
+                max_size = (228, 228)
+                if image.height > 228 or image.width > 228:
+                    image.thumbnail(max_size)
+                    image.save(self.avatar.path)
+            except Exception as e:
+                raise ValidationError('Image size error')
+
+
+class Team(models.Model):
+    class STATUS(TextChoices):
+        ACTIVE = 'active', 'Active'
+        AWAITING = 'awaiting', 'Awaiting'
+        DISBANED = 'disbanded', 'Disbanded'
+
+    name = models.CharField(
+        max_length=50,
+        unique=True,
+        verbose_name='Название команды',
+        )
+    members = models.ManyToManyField(
+        User,
+        through='TeamMembership',
+        related_name='teams',
+        verbose_name='Участники команды',
+    )
+    status = models.CharField(
+        max_length=10,
+        choices=STATUS.choices,
+        default=STATUS.AWAITING,
+        verbose_name='Статус команды',
+    )
+    description = models.CharField(
+        max_length=250,
+        default='',
+        blank=True,
+        verbose_name='Описание команды',
+    )
+    date_created = models.DateTimeField(
+        auto_now_add=True,
+        verbose_name='Дата создания команды',
+    )
+    date_disbanded = models.DateTimeField(
+        null=True,
+        blank=True,
+        verbose_name='Дата роспуска команды',
+    )
+    class Meta:
+        verbose_name='Команда'
+        verbose_name_plural='Команды'
+    
+    def add_member(self, user: User, reole: str = 'participant') -> None:
+        """ 
+        Добавляет пользователя в команду.
+        Если пользователь уже в команде, выбрасывает исключение.
+        """
+        if not TeamMembership.objects.filter(team=self, user=user).exists():
+            TeamMembership.objects.create(team=self, user=user, reole=reole)
+        raise ValidationError(
+            f'Пользователь {user.username} уже в команде'
+        )
+
+    def is_creator(self, user: User) -> bool:
+        """Проверяет, является ли пользователь создателем команды."""
+        try:
+            membership = TeamMembership.objects.get(team=self, user=user)
+            return membership.role == self.ROLECHOICES.CREATOR
+        except TeamMembership.DoesNotExist:
+            return False
+
+    def remove_member(
+            self,
+            user_to_remover: User,
+            current_user: User
+    ) -> None:
+        """
+        Удаляет пользователя из команды.
+        Если пользователь не в команде, выдает исключение.
+        Только для создателей команды.
+        """
+        if not self.is_creator(current_user):
+            raise ValidationError('Вы не являетесь создателем команды')
+        if user_to_remover == current_user:
```

Comment:
Check this before `not self.is_creator(...)`.

The `not self.is_creator(...)`  condition requires an additional db query. So, if the `remove_member` method receives an isntance of the `User` model through the `user_to_remove` and `current_user` fields, then the method will not execute a redundant db query.

Reply:
if we pass the queryset along with the user role, we will still send an additional request to the database. or I don't understand smth?

Reply:
nevermind i get it